shared/middleware/helpers/assets.rb

```diff
@@ -0,0 +1,68 @@
+#
+# Assets
+#
+class AssetBucket
+
+  ALLOWED_FILE_TYPES = %w(jpg jpeg gif png mp3)
+
+  def initialize
+    params = {region: 'us-east-1'}
+    if CDO.s3_access_key_id && CDO.s3_secret_access_key
+      params[:credentials] = Aws::Credentials.new(CDO.s3_access_key_id, CDO.s3_secret_access_key)
+    end
+    @s3 = Aws::S3::Client.new(params)
+  end
+
+  def list(encrypted_channel_id)
+    owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
+    prefix = "#{CDO.assets_s3_directory}/#{owner_id}/#{channel_id}"
+    @s3.list_objects(bucket:CDO.assets_s3_bucket, prefix:prefix).contents.map do |fileinfo|
+      filename = %r{#{prefix}/(.+)$}.match(fileinfo.key)[1]
+      mime_type = Sinatra::Base.mime_type(filename.split('.').last)
+      category = mime_type.split('/').first  # e.g. 'image' or 'audio'
+      {filename:filename, category:category, size:fileinfo.size}
```

Comment:
Josh and I discussed; Rubocop is a good plan (in a future CL) so that this part of the codebase is consistent with respect to space after colon.